### DISCUSSION TITLE
I Want to Find and Create TF1 Commands on My Own

### DISCUSSION THREAD

**Discussion by sson6203** [BrenekH/yamaha-rcp-docs#4] [General]
The main reason I'm curious about this is to "extract data" from products using Wireshark, especially when information is not readily available.

Recently, I practiced collecting data from the Yamaha TF1 using Wireshark.

As a result, I found that the data values for the 1ch on&off button and the 3ch on&off button were identical.

I couldn't figure out where the problem was.

I have one question.

What might I be doing wrong for the data to keep showing the same values?

Also, how did you manage to collect data that isn’t in the manual?

**Comment by BrenekH** [comment]
I assume you're talking about using the iPad app (or the TF Editor from a PC) to mute/unmute channels 1 and 3 correct? They shouldn't be using identical signals to do so. Do you mind sending a pcap file of the behavior you're seeing so I can look through it myself?

> Also, how did you manage to collect data that isn't in the manual?

It's been over a year since I started looking through all of this, but I never had to do a packet capture to collect the info. This was mostly because the basic structure was apparent from the official python scripts, and the massive list of possible options was in the [repository for Bitfocus Companion's Yamaha RCP module](https://github.com/bitfocus/companion-module-yamaha-rcp). If I wanted/needed to, I could absolutely use Wireshark on a TF1, and I've picked up traffic while diagnosing other problems, but there hasn't been a need for me to really dig deep in that way.

**Comment by sson6203** [comment]
I captured packets while operating the device with an iPad.

In other words, I used the iPad to toggle channel 1 on and off, and channel 3 on and off.

I don’t have the actual file you’re looking for since I no longer have the device on hand, but I do have a screenshot of the capture.

Red represents channel 1 being toggled on, and blue represents channel 3 being toggled on.

The reason I’m trying to collect data from the TF1 using Wireshark is that I want to understand the principles behind how IP-controlled devices work.

The reason I’m practicing with the Yamaha TF is that there are many successful examples to reference.

Could you analyze this using Wireshark and teach me how to successfully gather data?

![image](https://github.com/user-attachments/assets/5d661279-e0e1-45f3-849a-b395bacda541)

![image](https://github.com/user-attachments/assets/ee0fe8ff-27c0-4edc-b7f1-a0723fc6ab13)

**Reply by BrenekH** [reply]
Hmmm, that is rather odd. I would've expected to see a message in the data section similar to `set MIXER:Current/InCh/Fader/On 2 0 1` (channel 3 unmute), not `Mixing 1   Mixing`. I know I've seen a proper startup sequence between R Remote and a Tio-1608 D2 stage box in a packet capture with the correct commands, but I don't seem to have saved that session.

I think you're on the right track with this, but I would suggest maybe removing the filters and just going through the traffic for a bit to try and understand the full scope of everything, especially capture filters (not sure if you used them here, but in general I wouldn't recommend them unless you have good reason). Another possible issue is the method of capture. You said you were using an iPad, but how were you capturing the data? There are lots of ways to go about it, but the way that makes the most sense to me is either through a built-in capture function of your networking gear, or by setting up a port mirror using a smart switch to send to a Wireshark computer.

I will also say that the method of control will vary wildly from device to device. The TF protocol is rather simple because it's just TCP text, but other devices may use encryption or higher level protocols to transmit their data, which can make it harder to decipher. I would suggest learning about enterprise networking fundamentals (as if you were studying for a CompTIA exam) as that will also help you tremendously when doing deep dives like this, as well as asking for help in networking engineer forums.

**Comment by jskeet** [comment]
Just to explain why there isn't anything like `set MIXER:Current/InCh/Fader/On 2 0 1` - this is using the *binary* protocol (port 50368) rather than the simple text protocol on port 49280.

I found this discussion because I'm currently trying to reverse engineer the binary protocol for my DigiMixer project (https://codeblog.jonskeet.uk/category/digimixer/, https://github.com/jskeet/DemoCode/tree/main/DigiMixer)

In this case, the packets vary at byte 0x85 (although the real packet data starts at packet 0x36, the "MMIX" part) - changing channel 1 has a value of 0; changing channel 3 has a value of 2. I assume it's basically "channel number + 1" here.
